added sparkinfo command and logging

jbloom22 opened the pull request:
- number of partitions and storage level are logged for each command that requires vds

tpoterba commented on src/main/scala/org/broadinstitute/hail/driver/Command.scala:
```diff
@@ -249,8 +250,11 @@ abstract class Command {
-    else
+    else {
+      if (requiresVDS)
+        log.info(s"sparkinfo: ${state.vds.nPartitions} partitions, ${state.vds.rdd.getStorageLevel}")
```
add $name after sparkinfo -- will be a little clearer.

jbloom22 commented:
I added the command name, and switched to printing the readable string rather than the `StorageLevel(true, false, false, false, 1)` by adding a RichStorageLevel with toReadableString().  Also added a test in UtilsSuite

jbloom22 commented:
Rebased, back to you.

tpoterba commented:
Will merge when tests pass.  Make a post on discourse updates to introduce the new command!
